sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -203,14 +203,16 @@ case class DropTableCommand(
         case _ =>
       }
     }
-    try {
-      sparkSession.sharedState.cacheManager.uncacheQuery(sparkSession.table(tableName))
-    } catch {
-      case _: NoSuchTableException if ifExists =>
-      case NonFatal(e) => log.warn(e.toString, e)
+
+    if (!ifExists || catalog.tableExists(tableName)) {
```

Comment:
I feel it's more readable to write
```
if (catalog.tableExists(tableName)) {
  ... do the drop
} else if (ifExists) {
  // do nothing
} else {
  // log.warn("The table to drop does not exist: " +tableName)
}
```

Reply:
We can also update the UNCACHE TABLE, cc @gatorsmile

Reply:
I'll update soon.